tests/integration/test_remoted/test_agent_communication/test_shared_configuration.py

```diff
@@ -0,0 +1,126 @@
+                 For this purpose, the test will create an agent for each protocol within the module test cases. Then,
+                 it will try to send the shared configuration to the agent and then, check if the configuration is
+                 completely pushed.
+                 For example, if Wazuh Manager sends new shared files from group shared folder when the merged.mg
+                 checksum is received from an agent is different than the stored one.
+                 As the test has nothing to do with shared configuration files, we removed those rootcheck txt files
+                 from default agent group to reduce the time required by the test to make the checks.
+
+
+    parameters:
+        - test_configuration
+            type: dict
+            brief: Configuration applied to ossec.conf.
+        - test_metadata:
+            type: dict
+            brief: Test case metadata.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate all the log files and json alerts files before and after the test execution.
+        - configure_local_internal_options:
+            type: fixture
+            brief: Configure the Wazuh local internal options using the values from `local_internal_options`.
+        - daemons_handler:
+            type: fixture
+            brief: Restart service once the test finishes stops the daemons.
+        - simulate_agents
+            type: fixture
+            brief: create agents
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Apply changes to the ossec.conf configuration.
+    '''
+
+
+    agent = simulate_agents[0]
+
+    sender, injector = connect(agent = agent, protocol = test_metadata['protocol'])
+
+    wazuh_log_monitor = FileMonitor(WAZUH_LOG_PATH)
+
+    # Send the start-up message
+    sender.send_event(agent.startup_msg)
+
+    wazuh_log_monitor.start(callback=generate_callback(regex=patterns.START_UP, replacement={"agent_name": agent.name, "agent_ip": '127.0.0.1'}))
+
+    assert wazuh_log_monitor.callback_result
+
+
+    sender.send_event(agent.keep_alive_event)
+
+    time.sleep(5)
+
+    for pattern in test_metadata['patterns']:
+        check_queue_monitor(agent, pattern)
+
+
+    sender.send_event(agent.keep_alive_event)
+
+    # Check that push message doesn't appear again
+    with pytest.raises(AssertionError): #assert not callback_result
+        check_queue_monitor(agent, test_metadata['patterns'][0])
+        raise ValueError("Same shared configuration pushed twice!")
+
+
+    create_group('testing_group')
+    add_agent_to_group('testing_group', agent.id)
+
+    time.sleep(10)
+
+    wazuh_log_monitor.start(callback=generate_callback(regex=patterns.MERGED_NEW_SHARED_END_SEND))
+    assert wazuh_log_monitor.callback_result
+
+    check_queue_monitor(agent, test_metadata['patterns'][0])
+
+    delete_group('testing_group')
+    injector.stop_receive()
```

Comment:
Same, this should be the teardown.

Reply:
I spoke with Tomas and an agreement was reached not to do it due to how the tests were designed.